react/components/Upload/Upload.tsx

```diff
@@ -0,0 +1,185 @@
+import * as React from 'react'
+import { useMutation } from 'react-apollo'
+import { useDropzone } from 'react-dropzone'
+import { defineMessages, useIntl } from 'react-intl'
+import {
+  Button,
+  IconClose,
+  IconImage,
+  Input,
+  Spinner,
+  IconCaretDown,
+} from 'vtex.styleguide'
+
+import StyleButton from './StyleButton'
+import UploadFileMutation from '../../graphql/uploadFile.graphql'
+import { FormRawInputProps, InputTypes } from '../../typings/InputProps'
+import { HiddenInput } from '../Input'
+
+interface MutationData {
+  uploadFile: { fileUrl: string }
+}
+
+const MAX_SIZE = 4 * 1024 * 1024
+
+const messages = defineMessages({
+  addBtn: {
+    defaultMessage: 'Add',
+    id: 'admin/form.add-button',
+  },
+  addTitle: {
+    defaultMessage: 'Insert document',
+    id: 'admin/form.add-document.title',
+  },
+  addLabel: {
+    defaultMessage: 'Document URL',
+    id: 'admin/form.add-document.label',
+  },
+  addPlaceholder: {
+    defaultMessage: 'URL',
+    id: 'admin/form.add-document.placeholder',
+  },
+  fileSizeError: {
+    defaultMessage:
+      'File exceeds the size limit of 4MB. Please choose a smaller one.',
+    id: 'admin/form.document-uploader.error.file-size',
+  },
+  genericError: {
+    defaultMessage: 'Something went wrong. Please try again.',
+    id: 'admin/form.document-uploader.error.generic',
+  },
+  uploadBtn: {
+    defaultMessage: 'Upload document',
+    id: 'admin/store.upload-button',
+  },
+  uploadLabel: {
+    defaultMessage: 'Choose an document to upload',
+    id: 'admin/store.upload-document-label',
+  },
+})
+
+const InputUpload = (props: FormRawInputProps) => {
+  const intl = useIntl()
+  const [uploadFile] = useMutation<MutationData>(UploadFileMutation)
+  const ref = React.useRef<HTMLDivElement>(null)
+  const [isLoading, setIsLoading] = React.useState(false)
+  const [fileName, setFileName] = React.useState('')
+  const [isOpen, setIsOpen] = React.useState(false)
+  const [imageUrl, setImageUrl] = React.useState<string | undefined>()
+  const [error, setError] = React.useState<string | null>()
+
+  const { inputType = InputTypes.input, pointer, ...rest } = props
+
+  const onDropImage = async (files: any[]) => {
+    setError(null)
+
+    try {
+      if (files?.[0]) {
```

Comment:
This Component should catch only one file?!

Reply:
Yes!